src/contexts/addresses.tsx

```diff
@@ -210,7 +211,7 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
       if (!activeWalletMnemonic) throw new Error('Could not update address settings, mnemonic not found')
 
       if (!isPassphraseUsed)
-        storeAddressMetadataOfWallet(
+        AddressMetadataStorage.storeAddressMetadata(
```

Comment:
I will reply to EDIT2, because we both got to the same conclusion :D

That's exactly what I wanted to do. I wanted to override basically the `store` method of `PersistentEncryptedStorage` with a `store` method in `AddressMetadataStorage`, which would then call `super.store`. The problem is that when you want to override a class method, the params of that method need to be the same. Which is not the case here :/ Thus, I made the `store` method protected (so only subclasses can access it) and created `storeAddressMetadata`.

I am open to any other ideas... One idea is to rename the `store` method of `PersistentEncryptedStorage` to something like `_store`. Then, we are free to create `store` in `AddressMetadataStorage`. WDYT?

Reply:
👍 yes, your idea sounds like a fine compromise to me.